### PR TITLE
Add support for macOS(Darwin) / Windows

### PR DESCRIPTION
**Pull request by CoreNion**
References: #22 

Support for macOS was achieved by leveraging `sharedDarwinSource` and diverting the iOS source.

### CODE REVIEW

**Comment by chipweinberger** [comment]
yes please!

**Comment by jsiedentop** [comment]
Do i understand correctly that we are currently only waiting for a review by a verified googler like @jonasfj? I would be really grateful if someone could approve this pull request. Thank you very much!

File: src/webcrypto.h [hunk 1]
```diff
@@ -17,9 +17,13 @@
...
+#define WEBCRYPTO_EXPORT __declspec(dllexport)
+#else
+#define WEBCRYPTO_EXPORT                                                        \
   __attribute__((visibility("default"))) __attribute__((used))
+#endif
 
 // Function to lookup BoringSSL symbols based on index in the Sym enum.
 // See src/symbols.yaml for details.
-WEBCRYPTO_EXPORT void* webcrypto_lookup_symbol(int32_t index);
+WEBCRYPTO_EXPORT void* webcrypto_lookup_symbol(int32_t index);
```
Comment (jonasfj) [inline src/webcrypto.h:29]: Missing newline

File: windows/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,167 @@
+# The Flutter tooling requires that developers have a version of Visual Studio
```
Comment (jonasfj) [inline windows/CMakeLists.txt:1]: Missing license header

**Review by jonasfj — APPROVED** [review]
r+

**Comment by jonasfj** [comment]
For future reference please do consider making smaller PRs, and feel free to ping me or hit me up on discord. There is always too many things going on at the same time to keep track of everything, so stuff slips through the cracks.

**Comment by jonasfj** [comment]
@jsiedentop I think we should get this published, but I'm a bit concerned about releasing it without any tests. It just looks too good to be sitting around.

Please feel encouraged to contribute on https://github.com/google/webcrypto.dart/issues/76, I'm not on macos and debugging why the integration tests are hanging is not easy using Github Actions :D

**Pull request merged**